### ISSUE TITLE
Issue with reusability

### ISSUE THREAD

**Issue by cosminsg** [jburman/W8lessLabs.Blazor.LocalFiles#5]
Thank you for the code!
My goal is to allow the user to add multiple files not all at once, but subsequently. If the user forgets to add a file, he can add it later.
In order to achieve this goal, I have created a button that launches `
        fileSelect.SelectFiles(FilesSelected)
` from your code. From there, the files are added to a List of `SelectedFile` object. 
After the list is complete, the user would press a validation button that runs the logic for all the files in the List, including getting the filestream or the the filebytes.
However, if the user add new `SelectedFile` object to the list, the `fileSelect.GetFileReader(SelectedFile)` method no longer works because of a `System.NullReferenceException`. I think that somehow your method is binded to the initial SelectedFile[] so it only works if the list is not increased. 
There is no problem if the user removes a `SelectedFile` from the list. 
Is this a known issue or limitation? Can I get around it somehow?

Later edit:
For example, FileUpload in MatBlazor package has the following interface

public interface IMatFileUploadEntry
    {
        DateTime LastModified { get; }
        string Name { get; }
        long Size { get; }
        string Type { get; }
        **Task WriteToStreamAsync(Stream stream);**
    }

Basically, you can store the method related to the uploaded file, serving my purpose.

**Comment by jburman** [comment]
Ok, I haven't tried your scenario. The first thought I had would be to create a new blazor component that has the FileSelect component inside of it.  Then when your user clicks Add File you create another instance of your component.  That should keep the necessary file handles around until they are finished. Then you can iterate over all of your sub-components at the end and validate.

**Comment by jburman** [comment]
It looks like this will take some refactoring to support within the component itself.
I tested out a workaround and uploaded the code here:
https://github.com/jburman/W8lessLabs.Blazor.LocalFiles/blob/e1b6b209a473615c4c09eec32e8291665c37f190/test/W8lessLabs.Blazor.LocalFilesTest/Pages/Test.razor

Basically, it keeps a reference to the file selector around and creates a new one each time the user selects more files.

**Comment by jburman** [comment]
For the 1.0.0 release I added in a new FileSelectList component. This allows building up a list of selected files/removing them etc.
Example usage here: https://raw.githubusercontent.com/jburman/W8lessLabs.Blazor.LocalFiles/master/test/W8lessLabs.Blazor.LocalFilesTest/Pages/FileList.razor
